packages/appstore/src/components/cfds-listing/index.tsx

```diff
@@ -142,63 +132,19 @@ const CFDsListing = observer(() => {
-                            current_acc_status === 'proof_failed',
-                        ])
-                    ) {
-                        return MT5_ACCOUNT_STATUS.FAILED;
-                    } else if (
-                        getAuthStatus([
-                            poi_pending_for_bvi_labuan_vanuatu,
-                            current_acc_status === 'verification_pending',
-                        ])
-                    ) {
-                        return MT5_ACCOUNT_STATUS.PENDING;
-                    }
-                    return null;
-                }
-                case Jurisdiction.LABUAN: {
-                    if (
-                        getAuthStatus([
-                            poa_resubmit_for_labuan,
-                            is_idv_revoked,
-                            poi_resubmit_for_bvi_labuan_vanuatu,
-                            current_acc_status === 'proof_failed',
-                        ])
-                    ) {
-                        return MT5_ACCOUNT_STATUS.FAILED;
-                    } else if (
-                        getAuthStatus([
-                            poi_pending_for_bvi_labuan_vanuatu,
-                            current_acc_status === 'verification_pending',
-                        ])
-                    ) {
-                        return MT5_ACCOUNT_STATUS.PENDING;
-                    }
-                    return null;
-                }
-                default:
-                    if (current_acc_status === 'proof_failed') {
-                        return MT5_ACCOUNT_STATUS.FAILED;
-                    } else if (current_acc_status === 'verification_pending') {
-                        return MT5_ACCOUNT_STATUS.PENDING;
-                    } else if (current_acc_status === 'needs_verification') {
-                        return MT5_ACCOUNT_STATUS.NEEDS_VERIFICATION;
-                    } else if (current_acc_status === 'migrated_with_position') {
-                        return MT5_ACCOUNT_STATUS.MIGRATED_WITH_POSITION;
-                    } else if (current_acc_status === 'migrated_without_position') {
-                        return MT5_ACCOUNT_STATUS.MIGRATED_WITHOUT_POSITION;
-                    }
-                    return null;
+        } else if (jurisdiction) {
+            if (current_acc_status === 'proof_failed') {
+                return MT5_ACCOUNT_STATUS.FAILED;
+            } else if (current_acc_status === 'verification_pending') {
+                return MT5_ACCOUNT_STATUS.PENDING;
+            } else if (current_acc_status === 'needs_verification') {
+                return MT5_ACCOUNT_STATUS.NEEDS_VERIFICATION;
+            } else if (current_acc_status === 'migrated_with_position') {
+                return MT5_ACCOUNT_STATUS.MIGRATED_WITH_POSITION;
+            } else if (current_acc_status === 'migrated_without_position') {
+                return MT5_ACCOUNT_STATUS.MIGRATED_WITHOUT_POSITION;
             }
+            return null;
```

Comment:
will this be better? 

```suggestion
            switch (current_acc_status) {
                    case 'proof_failed':
                            return MT5_ACCOUNT_STATUS.FAILED;
                    case 'verification_pending':
                            return MT5_ACCOUNT_STATUS. PENDING;
                    case 'needs_verification':
                            return MT5_ACCOUNT_STATUS. NEEDS_VERIFICATION;
                    case 'migrated_with_position':
                            return MT5_ACCOUNT_STATUS. MIGRATED_WITH_POSITION;
                    case 'migrated_without_position':
                            return MT5_ACCOUNT_STATUS. MIGRATED_WITHOUT_POSITION;
                    default:
                            return null;
            }
```